### PR TITLE
Plane/Copter/Rover/Tracker: 4.6.0-beta1 release 

### PR DESCRIPTION
**Pull request by rmackay9**
This is the Copter, Rover, Tracker, Plane 4.6.0-beta1 release

To create the release notes I did this:

1. used Github to compile a full list of PRs in the ArduPilot-4.6 branch vs the Copter-4.5 branch.   The output from that can be found in [this spreadsheet](https://www.dropbox.com/scl/fi/yawmb18q7gwjpnses7shl/ardupilot-460-beta1.ods?rlkey=yq9ys5bldd8ona3shzyxx1t40&dl=0)
2. sorted the PRs by subsystem
3. manually reviewed each PR's title (and sometimes the content) in order to come up with a description of the change

Any and all feedback is greatly appreciated.  In particular I'd like to hear about:

a. changes that should be added or removed from the ReleaseNotes
b. corrections to change descriptions.  If possible please provide a copy-paste-able rewording
c. suggestions on re-ordering or moving changes within the ReleaseNotes

I haven't included Blimp or Sub but we could if maintainers of those vehicles would like (FYI @MichelleRos, @Williangalvani)

### CODE REVIEW

File: ArduCopter/ReleaseNotes.txt [hunk 1]
```diff
@@ -1,5 +1,432 @@
...
+15) ROS2 / DDS support
+
+- Airspeed published
+- Battery topic reports all available batteries
+- Compile-time configurable rates for each publisher
+- DDS_TIMEOUT_MS and DDS_MAX_RETRY set timeout and num retries when client pings XRCE agent
+- GPS global origin published at 1 Hz
+- High frequency raw imu data transmission
+- Joystick support
+- Support sending waypoints to Copter and Rover
```
Comment (Ryanf55) [inline ArduCopter/ReleaseNotes.txt:376]:
```suggestion
- Support sending waypoints to Copter and Rover
- Remove the XML refs file in favor of binary entity creation
```

From 296cdc58a3b43e8cbe199d6feef42bc583073702. This is an important breaking change.

**Review by Ryanf55 — COMMENTED** [review]
Nice to see these put together!

File: AntennaTracker/ReleaseNotes.txt [hunk 2]
```diff
@@ -1,5 +1,432 @@
...
+- "require" keyword works for modules in ROMFS (e.g. not on SD card)
+- revert_param.lua supports more params (e.g ATC_RATE_R/P/Y, PTCH2SRV_TCONST, RLL2SRV_TCONST, TECS_)
+- Scripts may receive mavlink messages which fail CRC (e.g messages which FC does not understand)
+- SD card formatting supported
+- Serial device loop back support (allows simulating a sensor using Lua)
+- set_target_throttle_rate_rpy allows rate control from scripts (new for Copter)
+- sitl_standby_sim.lua example shows how to switch between redundant flight controllers using an RC switch
+- Slung payload oscillation suppression applet added (see copter-slung-payload.lua)
+- Temperature sensor bindings added
+- uint64 support
```
Comment (tpwrules) [inline AntennaTracker/ReleaseNotes.txt:322]:
```suggestion
- uint64 support
- Various performance and memory usage optimizations
```

File: AntennaTracker/ReleaseNotes.txt [hunk 3]
```diff
@@ -1,5 +1,432 @@
...
+- Lua language checks improved (finds more errors)
+- MAVLink commands can be called from scripting
+- MCU voltage binding (see analog:mcu_voltage)
+- NMEA 2000 EFI driver (see EFI_NMEA2k.lua)
+- "open directory failed" false warning when scripts in ROMFS fixed
+- Param_Controller.lua supports quickly switching between 3 parameter sets via RC switch
+- Pass by reference values are always initialized
+- pelco_d_antennatracker.lua applet supports sending Pelco-D via RS-485 to PTZ cameras
+- plane_aerobatics.lua minor enhancements
+- REPL applet (read-evaluate-print-loop, see repl.lua)
```
Comment (tpwrules) [inline AntennaTracker/ReleaseNotes.txt:312]:
```suggestion
- REPL applet (read-evaluate-print-loop, see repl.lua) for interactive testing and experimentation
```

File: AntennaTracker/ReleaseNotes.txt [hunk 4]
```diff
@@ -1,5 +1,432 @@
...
+- "require" keyword works for modules in ROMFS (e.g. not on SD card)
```
Comment (tpwrules) [inline AntennaTracker/ReleaseNotes.txt:313]:
```suggestion
- "require" function failures in rare circumstances fixed
- "require" function works for modules in ROMFS (e.g. not on SD card)
```

File: AntennaTracker/ReleaseNotes.txt [hunk 5]
```diff
@@ -1,5 +1,432 @@
...
+- MAVLink via DroneCAN baud rate fix
+- SPI pins may also be used as GPIOs
+- Terrain cache size configurable (see TERRAIN_CACHE_SZ)
+
+18) Developer focused fixes and enhancements
+
+- AP_Camera gets example python code showing how to use GStreamer with UDP and RTSP video streams
+- Cygwin build fix for non-SITL builds
+- Cygwin build fixed with malloc wrap
+- DroneCAN and scripting support FlexDebug messages (see CAN_OPTION, FlexDebug.lua)
```
Comment (tpwrules) [inline AntennaTracker/ReleaseNotes.txt:421]:
```suggestion
- DroneCAN and scripting support FlexDebug messages (see CAN_Dn_UC_OPTION, FlexDebug.lua)
```

File: AntennaTracker/ReleaseNotes.txt [hunk 6]
```diff
@@ -1,5 +1,432 @@
...
+- Benewake RDS02U radar support
+- Bi-directional DShot on IOMCU supports reversible mask
+- Bi-directional DShot telemetry support on F103 8Mhz IOMCUs
+- BMM350 compass support
+- CAN rangefinders and proximity sensors may share a CAN bus (allows NRA24 and MR72 on a single CAN bus)
+- Compass calibration world magnetic model checks can use any position source (e.g. not just GPS)
+- CRSF baro and vertical speeed fixed
+- CRSF RX bind command support
+- DroneCAN battery monitor check to avoid memory corruption when type changed
+- DroneCAN DNA server fixes including removing use of invalid node IDs, optimise id allocation
```
Comment (tpwrules) [inline AntennaTracker/ReleaseNotes.txt:114]:
```suggestion
- DroneCAN DNA server fixes including removing use of invalid node IDs, faster ID allocation, elimination of rare inability to save info
```

File: AntennaTracker/ReleaseNotes.txt [hunk 7]
```diff
@@ -1,5 +1,432 @@
...
+- Param_Controller.lua supports quickly switching between 3 parameter sets via RC switch
+- Pass by reference values are always initialized
+- pelco_d_antennatracker.lua applet supports sending Pelco-D via RS-485 to PTZ cameras
+- plane_aerobatics.lua minor enhancements
+- REPL applet (read-evaluate-print-loop, see repl.lua)
+- "require" keyword works for modules in ROMFS (e.g. not on SD card)
+- revert_param.lua supports more params (e.g ATC_RATE_R/P/Y, PTCH2SRV_TCONST, RLL2SRV_TCONST, TECS_)
+- Scripts may receive mavlink messages which fail CRC (e.g messages which FC does not understand)
+- SD card formatting supported
+- Serial device loop back support (allows simulating a sensor using Lua)
```
Comment (tpwrules) [inline AntennaTracker/ReleaseNotes.txt:317]:
```suggestion
- Serial device simulation support (allows Lua to feed data to any supported serial protocol for e.g. sensor simulation)
```

**Review by tpwrules — APPROVED** [review]
Thanks so much, great to see the list so big!

Added some wording nits for some stuff I've worked on.

**Review by bnsgeyer — DISMISSED** [review]
I only have one change.
otherwise it looks good to me.

**Review by lthall — APPROVED** [review]

**Review by Hwurzburg — APPROVED** [review]

**Review by robertlong13 — APPROVED** [review]

**Review by andyp1per — APPROVED** [review]

File: ArduPlane/ReleaseNotes.txt [hunk 8]
```diff
@@ -1,5 +1,433 @@
...
+- BMM350 compass support
+- CAN rangefinders and proximity sensors may share a CAN bus (allows NRA24 and MR72 on a single CAN bus)
+- Compass calibration world magnetic model checks can use any position source (e.g. not just GPS)
+- CRSF baro and vertical speeed fixed
+- CRSF RX bind command support
+- DroneCAN battery monitor check to avoid memory corruption when type changed
+- DroneCAN DNA server fixes including removing use of invalid node IDs, faster ID allocation, elimination of rare inability to save info
+- DroneCAN EFI health check fix
+- DroneCAN ESC battery monitors calculate consumed mah
+- DroneCAN ESCs and servos forced to zero when disarmed
```
Comment (tridge) [inline ArduPlane/ReleaseNotes.txt:117]: only ESCs, servos work when disarmed

File: ArduPlane/ReleaseNotes.txt [hunk 9]
```diff
@@ -1,5 +1,433 @@
...
+- DroneCAN fix for deadlock and saturation of CAN bus
```
Comment (tridge) [inline ArduPlane/ReleaseNotes.txt:118]: this is in 4.5.7

**Review by tridge — CHANGES_REQUESTED** [review]
really well done on the release notes, just found some minor issues

File: ArduPlane/ReleaseNotes.txt (hunk 8 above)
Comment (rmackay9) [reply ArduPlane/ReleaseNotes.txt:117]: txs, fixed

File: ArduPlane/ReleaseNotes.txt (hunk 9 above)
Comment (rmackay9) [reply ArduPlane/ReleaseNotes.txt:118]: fixeed

File: ArduPlane/ReleaseNotes.txt [hunk 10]
```diff
@@ -1,5 +1,433 @@
...
+- ICE_PWM_IGN_ON, ICE_PWM_IGN_OFF, ICE_PWM_STRT_ON, ICE_PWM_STRT_OFF replaced with SERVOx_MIN/MAX/REVERSED (Plane only)
+- ICE_START_CHAN replaced with RC aux function (Plane only)
+- ICEngine retry max added (see ICE_STRT_MX_RTRY)
+- IE 2400 generator error message severity to GCS improved
+- INA2xx battery monitor support (reads temp, gets MAX_AMPS and SHUNT params)
+- MCP9600 temperature sensor I2C address fixed
+- MLX90614 temperature sensor support
+- MSP GPS ground course scaling fixed
+- MSP RC uses scaled RC inputs (fixes issue with RCx_REVERSED having no effect)
+- NanoRadar MR72 CAN radar support
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply ArduPlane/ReleaseNotes.txt:138]: txs, removed

File: ArduPlane/ReleaseNotes.txt [hunk 11]
```diff
@@ -1,5 +1,433 @@
...
+- UBlox GPS time ignored until at least 2D fix
+- VideoTX supports additional freq bands (RushFPV 3.3Ghz)
+- Volz logs desired and actual position, voltage, current, motor and PCB temp
+- Volz server feedback and logging fixed
+- Volz servo output in its own thread resulting in smoother movements
+- W25N02KV flash support
+
+4) Networking enhancements and fixes
+
+- Allow multiple UDP clients to connect/disconnect/reconnect
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply ArduPlane/ReleaseNotes.txt:162]: ok, removed

File: ArduPlane/ReleaseNotes.txt [hunk 12]
```diff
@@ -1,5 +1,433 @@
...
+- Siyi ZT6 and ZT30 thermal palette can be changed using camera-change-setting.lua script
+- Siyi ZT6 hardware id and set-lens fixed
+- Topotek gimbal support
+- Trigger distance ignores GPS status and only uses EKF reported location
+
+6) Harmonic notch enhancements
+
+- Harmonic notch filter freq clamping and disabling reworked
+- Harmonic notch handles negative ESC RPMs
+- Harmonic notch supports per-motor throttle-based harmonic notch
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply ArduPlane/ReleaseNotes.txt:188]: ok, I've added a new line, "- Harmonic notch is active in forward flight on quadplanes"

File: ArduPlane/ReleaseNotes.txt [hunk 13]
```diff
@@ -1,5 +1,433 @@
...
+- Quadplane assistance check enhancements
+- Quadplane Deca frame support
+- Quadplane gets smoother takeoff by input shaping target accel and velocity
+- Servo wiggles in altitude wait staged to be one after another
+- Set_position_target_global_int accepts MAV_FRAME_GLOBAL_RELATIVE_ALT and MAV_FRAME_GLOBAL_TERRAIN_ALT frames
+- Takeoff airspeed control improved (see TKOFF_MODE, TKOFF_THR_MIN)
+- Takeoff fixes for fence autoenable
+- Takeoff improvements including less overshoot of TKOFF_ALT
+- TECS reset along with other controllers (important if plane dropped from balloon)
+- Throttle passthrough support while in VTOL modes removed
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply ArduPlane/ReleaseNotes.txt:255]: ok, removed

File: ArduPlane/ReleaseNotes.txt [hunk 14]
```diff
@@ -1,5 +1,433 @@
...
+- Various performance and memory usage optimizations
+- VTOL-quicktune.lua minor enhancements including increasing YAW_FLTE_MAX to 8
+- x-quad-cg-allocation.lua applet corrects for the CoM discrepancy in a quad-X frame
+
+13) GCS / mavlink related changes and fixes
+
+- BATTERY2 message deprecated (replaced by BATTERY_STATUS)
+- CMD_MISSION_START/STOP rejected if first-item/last-item args provided
+- Deny attempts to upload two missions simultaneously
+- ESC telemetry works even when less than 4 ESCs
```
[the remark this answers is not in the document]
Comment (rmackay9) [reply ArduPlane/ReleaseNotes.txt:332]: txs removed

**Pull request merged**